pkg/storage/replica_command.go

```diff
@@ -1565,6 +1565,8 @@ func evalPushTxn(
 	if reply.PusheeTxn.Epoch < args.PusheeTxn.Epoch {
 		reply.PusheeTxn.Epoch = args.PusheeTxn.Epoch
 	}
+	reply.PusheeTxn.UpgradePriority(args.PusheeTxn.Priority)
```

Comment:
Hmm, it seems we cannot use `Transaction.Update` here since the function takes `*Transaction` while the type of`args.PusheeTxn` is `*enginepb.TxnMeta`.

I've moved the code block as suggested.